File: programmers/python 3/42842.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42842

def solution(brown, yellow):
    h = 3
    while True:
        w = (brown - 2 * h + 4) // 2
        if w * h == brown + yellow:
            return [w, h]
        h += 1

File: programmers/swift/42842.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42842

func solution(_ brown: Int, _ yellow: Int) -> [Int] {
    var (width, height) = (0, 2)
    while width * height != brown + yellow {
        height += 1
        width = (brown - 2 * height + 4) / 2
    }

    return [width, height]
}